kernel/rros/thread.rs

```diff
@@ -1739,8 +1884,9 @@ fn thread_factory_build(
     // 	strncpy(tsk->comm, rros_element_name(&curr->element),
     // 		sizeof(tsk->comm));
     // 	tsk->comm[sizeof(tsk->comm) - 1] = '\0';
-    // `RrosElement.pointer` point to `rros_thread` struct.
-    let rros_thread_ptr: *mut RrosThread = curr.locked_data().get();
+    // `RrosElement.pointer` point to `RrosThread` struct.
+    let rros_thread_ptr =
+        Arc::as_ptr(&curr) as *const SpinLock<RrosThread> as *mut SpinLock<RrosThread>;
     (curr.lock().element.borrow_mut()).pointer = rros_thread_ptr as *mut u8;
```

Comment:
Does it have to be `*mut SpinLock<RrosThread>` rather than `*mut RrosThread`?

Reply:
I think this code must be implemented like this. `element.pointer` is the only value that could get back the element itself. If we use `*mut RrosThraed`, it will casue a memory leak on the `Spinlock`, and we can't get back the `Spinlock`.